content/posts/Aman_Asrani_phase_1_report_Enhancing_Circuitverse_User_Experiance_GSOC2024.md

```diff
@@ -0,0 +1,166 @@
+---
+title: "Enhancing CircuitVerse User Experience | GSOC2024 | Phase 1 Report"
+date: 2024-07-12T02:00:03+05:30
+draft: false
+author: "Aman Asrani"
+type: post
+---
+
+![GSOC2024_Phase_1](/images/Aman_Asrani/Banner.jpg)
+
+Hii Everyone,
+
+Welcome to this blog, dedicated to summarizing the achievements of the first phase of Google Summer of Code 2024 for the project **Enhancing CircuitVerse User Experiance**
+
+### About the Project 🖥
+
+> **[_Enhancing CircuitVerse User Experiance_](https://summerofcode.withgoogle.com/programs/2024/projects/h3QKa7eD)
+> The "Enhancing CircuitVerse User Experience" project is all about making CircuitVerse.org more engaging and user-friendly for everyone who loves digital circuits by introducing many features such improving SSO integration,better learning management system (LMS) and single sign-on (SSO) integration, plus email verification for extra security, Logs Mtransitioning to more efficient UI components will help create a smoother and safer user experience,Weekly Contest, Logs Maintenance, UI-Based Language Contribution**
+
+So far, we have covered and implemented the following features in the first phase of the project:
+
+- SAML Based SSO
+- Email Verification During Registration for New Users 
+- Send Email Verification button
+- POC for Email Verification for Pre-Existing Users
+- Migration to View Components
+- Logs Maintenance
+
+### SAML based SSO
+
+Single sign-on (SSO) is an identification method that enables users to log in to multiple applications and websites with one set of credentials. SSO streamlines the authentication process for users.SAML (Security Assertion Markup Language) is a markup language designed for exchanging authentication information between the user, the identity provider, and the service provider.
+
+I initially encountered a number of problems some of which are because of gem `devise-saml-authenticable` and  gem `ruby-saml` versioning compatibility. Finally,  I was able to redirect to the OKTA login page and obtain the SAML answer after addressing these. But then I ran into a problem while trying to sign the user which took too much time and then finally, during a meeting with Tanmoy Bhaiya, we discovered that the create method in the `saml_sessions_controller` was being overridden by the `create` method from the `devise_saml_authenticable` gem. Then finally SAML based SSO works smoothly.
```

Comment:
@Asrani-Aman Let's use "we" rather than "I", this applies to all content.
This blog is from our as a project team side to the community.

Reply:
I am really sorry, I shouldn't use "I"